### PR TITLE
DOC: Improved docs for `jnp.tri`, `jnp.tril` and `jnp.triu`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -4710,17 +4710,101 @@ def trapezoid(y: ArrayLike, x: ArrayLike | None = None, dx: ArrayLike = 1.0,
...
 
-@util.implements(np.tri)
 def tri(N: int, M: int | None = None, k: int = 0, dtype: DTypeLike | None = None) -> Array:
+  r"""Return an array with ones on and below the diagonal and zeros elsewhere.
+
+  JAX implementation of :func:`numpy.tri`
+
+  Args:
+    N: int. Dimension of the rows of the returned array.
+    M: optional, int, default=None. Dimension of the columns of the returned
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:4720]: I don't think `default=None` is meaningful here.

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -4710,17 +4710,101 @@ def trapezoid(y: ArrayLike, x: ArrayLike | None = None, dx: ArrayLike = 1.0,
...
 
-@util.implements(np.tril)
 @partial(jit, static_argnames=('k',))
 def tril(m: ArrayLike, k: int = 0) -> Array:
+  r"""Return lower triangle of an array.
+
+  JAX implementation of :func:`numpy.tril`
+
+  Args:
+    m: input array.
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:4776]: Mention that this should be at least two-dimensional.

File: jax/_src/numpy/lax_numpy.py [hunk 3]
```diff
@@ -4710,17 +4710,101 @@ def trapezoid(y: ArrayLike, x: ArrayLike | None = None, dx: ArrayLike = 1.0,
...
+  Returns:
+    An array containing the lower triangle of the given array with elements above
+    the sub-diagonal specified by ``k`` are set to zero.
+
+  See also:
+    - :func:`jax.numpy.triu`: Returns an upper triangle of an array.
+    - :func:`jax.numpy.tri`: Returns an array with ones on and below the diagonal
+      and zeros elsewhere.
+
+  Examples:
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:4791]: We should give an example where the input has more than two dimensions (in which case `jnp.tril` operates batch-wise on the trailing axes)

File: jax/_src/numpy/lax_numpy.py [hunk 4]
```diff
@@ -4730,9 +4814,49 @@ def tril(m: ArrayLike, k: int = 0) -> Array:
   return lax.select(lax.broadcast(mask, m_shape[:-2]), m, zeros_like(m))
 
 
-@util.implements(np.triu, update_doc=False)
 @partial(jit, static_argnames=('k',))
 def triu(m: ArrayLike, k: int = 0) -> Array:
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:4818]: Same comments here as in `tril` above.

File: jax/_src/numpy/lax_numpy.py [hunk 5]
```diff
@@ -4710,17 +4710,101 @@ def trapezoid(y: ArrayLike, x: ArrayLike | None = None, dx: ArrayLike = 1.0,
...
+
+  Args:
+    m: input array.
+    k: k: optional, int, default=0. Specifies the sub-diagonal above which the
+      elements of the array are set to zero. ``k=0`` refers to main diagonal,
+      ``k<0`` refers to sub-diagonal below the main diagonal and ``k>0`` refers
+      to sub-diagonal above the main diagonal.
+
+  Returns:
+    An array containing the lower triangle of the given array with elements above
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:4783]: An array with the same shape as the input

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Removed it.

File: jax/_src/numpy/lax_numpy.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Mentioned: `Must have "m.ndim >= 2"`

File: jax/_src/numpy/lax_numpy.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py:4790]: Included one example with three dimensional array for both `jnp.tril` and `jnp.triu`

File: jax/_src/numpy/lax_numpy.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py:4830]: Done

File: jax/_src/numpy/lax_numpy.py (hunk 5 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Done

File: jax/_src/numpy/lax_numpy.py [hunk 6]
```diff
@@ -4773,15 +4773,14 @@ def tril(m: ArrayLike, k: int = 0) -> Array:
...
+    m: input array. Must have ``m.ndim >= 2``.
     k: k: optional, int, default=0. Specifies the sub-diagonal above which the
       elements of the array are set to zero. ``k=0`` refers to main diagonal,
       ``k<0`` refers to sub-diagonal below the main diagonal and ``k>0`` refers
       to sub-diagonal above the main diagonal.
 
   Returns:
-    An array containing the lower triangle of the given array with elements above
-    the sub-diagonal specified by ``k`` are set to zero.
+    An array with the same shape as the input
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:4783]: I meant here that you should keep the original description, but also mention the shape of the output.

File: jax/_src/numpy/lax_numpy.py [hunk 7]
```diff
@@ -4821,15 +4833,14 @@ def triu(m: ArrayLike, k: int = 0) -> Array:
...
+    m: input array. Must have ``m.ndim >= 2``.
     k: k: optional, int, default=0. Specifies the sub-diagonal below which the
       elements of the array are set to zero. ``k=0`` refers to main diagonal,
       ``k<0`` refers to sub-diagonal below the main diagonal and ``k>0`` refers
       to sub-diagonal above the main diagonal.
 
   Returns:
-    An array containing the upper triangle of the given array with elements below
-    the sub-diagonal specified by ``k`` are set to zero.
+    An array with the same shape as the input
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:4843]: Same here

File: jax/_src/numpy/lax_numpy.py (hunk 6 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Sorry, I misunderstood it. Changed it.

File: jax/_src/numpy/lax_numpy.py (hunk 7 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/lax_numpy.py]: Done

**Review by jakevdp — COMMENTED** [review]
Looks good, thanks! Please squash changes, and then we can get this merged 👍

**Comment by rajasekharporeddy** [comment]
Squashed into single commit.

**Review by jakevdp — APPROVED** [review]

**Pull request merged**